crates/bevy_ecs/src/system/builder.rs

```diff
@@ -337,6 +339,194 @@ unsafe impl<'s, T: FromWorld + Send + 'static> SystemParamBuilder<Local<'s, T>>
     }
 }
 
+/// A [`SystemParamBuilder`] for a [`FilteredResources`].
+/// See the [`FilteredResources`] docs for examples.
+pub struct FilteredResourcesParamBuilder<T>(T);
+
+impl<T> FilteredResourcesParamBuilder<T> {
+    /// Creates a [`SystemParamBuilder`] for a [`FilteredResources`] that accepts a callback to configure the [`FilteredResourcesBuilder`].
+    pub fn new(f: T) -> Self
+    where
+        T: FnOnce(&mut FilteredResourcesBuilder),
+    {
+        Self(f)
+    }
+}
+
+impl<'a> FilteredResourcesParamBuilder<Box<dyn FnOnce(&mut FilteredResourcesBuilder) + 'a>> {
+    /// Creates a [`SystemParamBuilder`] for a [`FilteredResources`] that accepts a callback to configure the [`FilteredResourcesBuilder`].
+    /// This boxes the callback so that it has a common type.
+    pub fn new_box(f: impl FnOnce(&mut FilteredResourcesBuilder) + 'a) -> Self {
+        Self(Box::new(f))
+    }
+}
+
+// SAFETY: Calls `Access::build`.
+unsafe impl<'w, T: FnOnce(&mut FilteredResourcesBuilder)> SystemParamBuilder<FilteredResources<'w>>
+    for FilteredResourcesParamBuilder<T>
+{
+    fn build(
+        self,
+        world: &mut World,
+        system_meta: &mut SystemMeta,
+    ) -> <FilteredResources<'w> as SystemParam>::State {
+        let mut builder = FilteredResourcesBuilder::new(world);
+        (self.0)(&mut builder);
+        let access = builder.build();
+        SystemParamBuilder::<FilteredResources>::build(access, world, system_meta)
+    }
+}
+
+// SAFETY: Resource ComponentId and ArchetypeComponentId access is applied to SystemMeta. If this FilteredResources
+// conflicts with any prior access, a panic will occur.
+unsafe impl<'w> SystemParamBuilder<FilteredResources<'w>> for Access<ComponentId> {
```

Comment:
This `impl` should be for `FilteredResourcesBuilder`

Reply:
There's one for that, too!  I thought it might sometimes be useful to be able to run the builder earlier and just pass the `Access` instead of needing a closure.  

I don't actually have a use case, though, and I could remove it if you think that would be more clear.  

(You can build a `Query` by passing a `QueryState` directly, but that one has more interesting state.)

Reply:
Ah right, I thought it was a different impl.

Your usecase seems reasonable, but I would avoid exposing such an impl directly on `Access<ComponentId>` and instead make a separate `FilteredResourcesState` that wraps it.

Reply:
Yeah, you're right that it should be wrapped if we expose it.  Since the use case is just hypothetical, I'll just remove the impl rather than introduce even more types.